compiler-cli/src/publish.rs

```diff
@@ -754,3 +752,113 @@ fn prevent_publish_git_dependency() {
 fn quotes(x: &str) -> String {
     format!(r#"<<"{x}">>"#)
 }
+
+#[test]
+fn exported_project_files_test() {
+    let tmp = tempfile::tempdir().unwrap();
+    let path = Utf8PathBuf::from_path_buf(tmp.path().join("my_project")).expect("Non Utf8 Path");
+
+    let exported_project_files = &[
+        "LICENCE",
+        "LICENCE.md",
+        "LICENCE.txt",
+        "LICENSE",
+        "LICENSE.md",
+        "LICENSE.txt",
+        "NOTICE",
+        "NOTICE.md",
+        "NOTICE.txt",
+        "README",
+        "README.md",
+        "README.txt",
+        "gleam.toml",
+        "priv/wibble",
+        "priv/wobble.js",
+        "src/exported.gleam",
+        "src/exported_ffi.erl",
+        "src/exported_ffi.ex",
+        "src/exported_ffi.hrl",
+        "src/exported_ffi.js",
+        "src/exported_ffi.mjs",
+        "src/exported_ffi.ts",
+        "src/nested/exported.gleam",
+        "src/nested/exported_ffi.erl",
+        "src/nested/exported_ffi.ex",
+        "src/nested/exported_ffi.hrl",
+        "src/nested/exported_ffi.js",
+        "src/nested/exported_ffi.mjs",
+        "src/nested/exported_ffi.ts",
+    ];
+
+    let unexported_project_files = &[
+        ".git/",
+        ".github/workflows/test.yml",
+        ".gitignore",
+        "build/",
+        "ignored.txt",
+        "priv/ignored",
+        "src/.hidden/hidden.gleam",
+        "src/.hidden/hidden_ffi.erl",
+        "src/.hidden/hidden_ffi.mjs",
+        "src/.hidden.gleam",
+        "src/.hidden_ffi.erl",
+        "src/.hidden_ffi.mjs",
```

Comment:
It appears that any files and folders prefixed with a dot `.` are ignored regardless of `.gitignore`. This is a setting that we can toggle on `ignore::WalkDir`. Should we include those as well?

Reply:
I don't think we should be using `ignore` for this as we don't want to ignore anything.

Reply:
Pardon? I thought we would be ignoring .gitignored files.

What I'm asking is if we should ignore files and directories starting with a dot as well (currently they are ignored, regardless of .gitignore). I'd say that perhaps we shouldn't.

Reply:
See my comment here: https://github.com/gleam-lang/gleam/pull/4256#discussion_r1976389845

Reply:
Oh okay! I hadn't seen it, sorry.

We can still use `ignore::WalkDir` but disabling all filters, that should work (I'm using this for `erlang_files`).

Alternatively, we can try to use our existing `DirWalker` implementation, but it requires a reference to a `FilesystemReader`, so we can't just create one in the function as it'd still be borrowed, requiring some way to pass one as a parameter everywhere, which was why I avoided it for now... Let me know if we should find a way to do this instead.

Reply:
Okay, I've disabled all filters for `ignore::WalkDir` which seems to do the job.

Note that this also affects the formatter, as it uses `fs::gleam_files`. It will now format `.gitignored` files as well.

Reply:
Ah! Thank you!